// way2go-authentication/lib/components/Login/LoginApi.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _axios = require('axios');

var _axios2 = _interopRequireDefault(_axios);

var _apiUrls = require('../Constants/apiUrls');

var ApiUrls = _interopRequireWildcard(_apiUrls);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var LoginApi = function () {
  function LoginApi() {
    _classCallCheck(this, LoginApi);
  }

  _createClass(LoginApi, null, [{
    key: 'login',
    value: function login(loginObj) {
      var reqObj = {
        method: 'POST',
        url: ApiUrls.authservice + '/auth/login',
        data: loginObj
      };
      return (0, _axios2.default)(reqObj);
    }
  }, {
    key: 'authenticateService',
    value: function authenticateService(orgId) {
      var reqObj = {
        method: 'GET',
        url: ApiUrls.authservice + '/auth/getAuthConfig/' + orgId
      };
      return (0, _axios2.default)(reqObj);
    }
  }]);

  return LoginApi;
}();

exports.default = LoginApi;
