// way2go-authentication/lib/components/Login/LoginComponent.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.LoginComponent = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _colors = require('material-ui/styles/colors');

var _MuiThemeProvider = require('material-ui/styles/MuiThemeProvider');

var _MuiThemeProvider2 = _interopRequireDefault(_MuiThemeProvider);

var _getMuiTheme = require('material-ui/styles/getMuiTheme');

var _getMuiTheme2 = _interopRequireDefault(_getMuiTheme);

var _TextField = require('material-ui/TextField');

var _TextField2 = _interopRequireDefault(_TextField);

var _RaisedButton = require('material-ui/RaisedButton');

var _RaisedButton2 = _interopRequireDefault(_RaisedButton);

var _Paper = require('material-ui/Paper');

var _Paper2 = _interopRequireDefault(_Paper);

var _container = require('muicss/lib/react/container');

var _container2 = _interopRequireDefault(_container);

var _row = require('muicss/lib/react/row');

var _row2 = _interopRequireDefault(_row);

var _col = require('muicss/lib/react/col');

var _col2 = _interopRequireDefault(_col);

var _web_logoback = require('../../assets/images/web_logoback.png');

var _web_logoback2 = _interopRequireDefault(_web_logoback);

var _login_web = require('../../assets/images/login_web.jpg');

var _login_web2 = _interopRequireDefault(_login_web);

var _Logo = require('./Logo');

var _Logo2 = _interopRequireDefault(_Logo);

var _forgotPwdComponent = require('../ForgotPassword/forgotPwdComponent');

var _forgotPwdComponent2 = _interopRequireDefault(_forgotPwdComponent);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// import {errormsg} from '../../css/style.css';
// import {errormsg} from '../../css/style.css';
var muiTheme = (0, _getMuiTheme2.default)({
    palette: {
        primary1Color: _colors.purple500
    }
});

var style = {
    padding: 20,
    display: 'inline-block',
    borderRadius: '6px',
    border: '1px solid #dbdbdb',
    backgroundColor: 'rgb(245, 245, 245)',
    width: '263px',
    position: 'relative',
    marginTop: '-133px',
    paddingTop: '125px'
};

var backgroundStyle = {
    backgroundImage: 'url("' + _login_web2.default + '")',
    WebkitBackgroundSize: 'cover',
    height: '100%'
};
var mainDivStyle = {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    height: '100%'
};
var logoDivStyle = {
    width: '285px',
    backgroundImage: 'url("' + _web_logoback2.default + '")',
    position: 'relative',
    zIndex: 150,
    height: '77px',
    marginTop: '56px',
    marginLeft: 'auto',
    marginRight: 'auto'
};
var LoginComponent = exports.LoginComponent = function LoginComponent(_ref) {
    var loginObj = _ref.loginObj,
        loginSubmit = _ref.loginSubmit,
        loginResponseObj = _ref.loginResponseObj,
        onChange = _ref.onChange,
        showLoggedOut = _ref.showLoggedOut;


    return _react2.default.createElement(
        'div',
        { style: backgroundStyle },
        _react2.default.createElement(
            _container2.default,
            { fluid: true },
            _react2.default.createElement(
                _row2.default,
                { style: mainDivStyle },
                _react2.default.createElement(
                    _col2.default,
                    { 'md-offset': '6', md: '6', 'sm-offset': '4', sm: '8', className: 'mui--text-center' },
                    _react2.default.createElement(
                        'div',
                        { style: logoDivStyle },
                        _react2.default.createElement(_Logo2.default, null)
                    ),
                    _react2.default.createElement(
                        _MuiThemeProvider2.default,
                        { muiTheme: muiTheme },
                        _react2.default.createElement(
                            _Paper2.default,
                            { style: style, zDepth: 1 },
                            _react2.default.createElement(
                                'form',
                                { id: 'loginComponent' },
                                _react2.default.createElement(_TextField2.default, { name: 'userName', floatingLabelText: 'Username', defaultValue: loginObj.userName, onChange: onChange, fullWidth: true, autoFocus: true }),
                                _react2.default.createElement('br', null),
                                _react2.default.createElement(_TextField2.default, { name: 'password', floatingLabelText: 'Password', defaultValue: loginObj.password, type: 'password', onChange: onChange, fullWidth: true }),
                                _react2.default.createElement('br', null),
                                _react2.default.createElement('br', null),
                                showLoggedOut && _react2.default.createElement(
                                    'div',
                                    { className: 'mui--text-caption', style: { color: 'red' } },
                                    _react2.default.createElement('br', null),
                                    'You have logged out successfully',
                                    _react2.default.createElement('br', null),
                                    _react2.default.createElement('br', null)
                                ),
                                _react2.default.createElement(
                                    'div',
                                    { style: { padding: '8px 16px' } },
                                    _react2.default.createElement(_RaisedButton2.default, { type: 'submit', label: 'Log In', primary: true, onClick: loginSubmit, fullWidth: true })
                                ),
                                _react2.default.createElement('br', null),
                                _react2.default.createElement(
                                    'div',
                                    { style: { cursor: 'pointer' }, onClick: function onClick() {
                                            document.getElementById("loginComponent").style.display = 'none';document.getElementById("forgotPwdComponent").style.display = 'block';document.getElementById("email").value = '';document.getElementById("userName").value = '';
                                        } },
                                    _react2.default.createElement(
                                        'u',
                                        null,
                                        'Forgot Password'
                                    )
                                ),
                                _react2.default.createElement('br', null),
                                ' ',
                                loginResponseObj.outputCode == 500 ? _react2.default.createElement(ShowError, { error: loginResponseObj }) : _react2.default.createElement(
                                    'span',
                                    null,
                                    '\xA0'
                                )
                            ),
                            _react2.default.createElement(
                                'div',
                                { id: 'forgotPwdComponent', style: { display: 'none' } },
                                _react2.default.createElement(_forgotPwdComponent2.default, null)
                            )
                        )
                    )
                )
            )
        )
    );
};

LoginComponent.propTypes = {
    loginObj: _propTypes2.default.object.isRequired,
    loginSubmit: _propTypes2.default.func.isRequired,
    onChange: _propTypes2.default.func.isRequired,
    loginResponseObj: _propTypes2.default.object.isRequired,
    onFPChange: _propTypes2.default.func.isRequired
};

var ShowError = function ShowError(_ref2) {
    var error = _ref2.error;

    return _react2.default.createElement(
        'div',
        { className: 'errormsg' },
        error.message
    );
};

ShowError.propTypes = {
    error: _propTypes2.default.object.isRequired
};

exports.default = LoginComponent;
