// way2go-common/lib/components/Themes/ThemePalettes.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _colors = require('material-ui/styles/colors');

var colors = _interopRequireWildcard(_colors);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

var ThemePalettes = function ThemePalettes(color) {

    var colorName = color.replace(/-([a-z])/g, function (g) {
        return g[1].toUpperCase();
    }); //color name in camel case

    switch (color) {

        default:
            return {
                palette: {
                    primary1Color: colors[colorName + '500'],
                    primary2Color: colors[colorName + '700'],
                    primary3Color: colors.grey600,
                    primaryDarkColor: colors[colorName + '800'] || colors['cyan800'],
                    primaryLightColor1: colors[colorName + '50'],
                    primaryLightColor2: colors[colorName + '100'],
                    alternateTextColor: '#FFF',
                    accent3Color: colors['grey700']
                }
            };
    }
};

exports.default = ThemePalettes;
