// way2go-common/lib/components/Storage/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _jsCookie = require('js-cookie');

var _jsCookie2 = _interopRequireDefault(_jsCookie);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var storage = function storage() {
    var storage = {};
    storage.options = {
        'AUTH_TOKEN': { key: 'authtoken' },
        'ORG_ID': { key: 'orgId', defaultValue: 1001 },
        'LOCALE': { key: 'locale', defaultValue: 'en-us' },
        'THEME': { key: 'theme', defaultValue: { primary: 'primaryPalette', secondary: 'secondaryPalette', warn: 'warnPalette' } },
        'THEME_PRIMARY': { key: 'primary', defaultValue: 'primaryPalette' },
        'THEME_SECONDARY': { key: 'secondary', defaultValue: 'secondaryPalette' },
        'THEME_WARNING': { key: 'warn', defaultValue: 'warnPalette' },
        'THEME_BACKGROUND': { key: 'background', defaultValue: 'grey' },
        'USER_ROLES': { key: 'ur' },
        'DELEGATOR': { key: 'delegator' },
        'USER_ID': { key: 'userId' },
        'USERNAME': { key: 'un' },
        'USER_DETAILS': { key: 'ud' }
    };

    storage.cookieExpiry = 7;
    storage.getValue = function (option) {
        if (this.isLocalStorageSupported()) {
            return localStorage.getItem(option.key) ? JSON.parse(localStorage.getItem(option.key)) : option.defaultValue;
        } else {
            return _jsCookie2.default.get(option.key) ? JSON.parse(_jsCookie2.default.get(option.key)) : option.defaultValue;
        }
    };

    storage.setValue = function (option, value, storeAsString) {
        if (this.isLocalStorageSupported()) {
            if (isNaN(value) || storeAsString) {
                localStorage.setItem(option.key, JSON.stringify(value));
            } else {
                localStorage.setItem(option.key, value);
            }
        } else {
            _jsCookie2.default.set(option.key, JSON.stringify(value), { expires: this.cookieExpiry });
        }
    };

    storage.removeKey = function (option) {
        if (this.isLocalStorageSupported()) {
            delete localStorage[option.key];
        } else {
            _jsCookie2.default.remove(option.key);
        }
    };

    storage.cleanStorage = function () {
        if (this.isLocalStorageSupported()) {
            var key = null;
            for (key in localStorage) {
                if (key != this.options.ORG_ID.key) {
                    // except for orgId remove all cookies
                    delete localStorage[key];
                }
            }
        } else {
            var cookieList = _jsCookie2.default.get();
            for (var cookie in cookieList) {
                if (cookie != "orgId") {
                    // except for orgId remove all cookies
                    _jsCookie2.default.remove(cookie);
                }
            }
        }
    };

    // To get the Auth Token
    storage.getAuthToken = function () {
        try {
            return this.getValue(this.options.AUTH_TOKEN);
        } catch (e) {
            alert('getAuthToken' + e.message);
        }
    };

    // Check if local storage is supported
    storage.isLocalStorageSupported = function () {
        try {
            var test = localStorage;

            return true;
        } catch (e) {
            return false;
        }
    };

    // Check if local storage is supported
    storage.isAuthenticated = function () {
        try {
            var result = this.getValue(this.options.AUTH_TOKEN);
            if (result) return true;else return false;
        } catch (e) {
            return false;
        }
    };

    return storage;
};

exports.default = storage;
