// way2go-common/src/components/AlertService/AlertService.js
import React from 'react';
import Snackbar from 'material-ui/Snackbar';
import muiThemeable from 'material-ui/styles/muiThemeable';

class AlertService extends React.Component {
  constructor(props) {
    super(props);

    this.state = {open: false, message: ""};

    this.handleRequestClose = this.handleRequestClose.bind(this);

    this.timer = undefined;
  }

  componentDidMount(){
    this.setState({message: this.props.message, open: true});
  }

  componentWillReceiveProps(nextProps) {
    this.setState({
      open: true,
      message: nextProps.message
    });
  }

  handleRequestClose() {
    this.setState({open: false});
    if(this.props.handleRequestClose)
      this.props.handleRequestClose();
  };

  render() {
    return (
      <Snackbar
          bodyStyle={{backgroundColor: this.props.muiTheme.palette.primaryDarkColor}}
          open={this.state.open}
          message={this.state.message}
          autoHideDuration={4000}
          onRequestClose={this.handleRequestClose}
        />
    );
  }
}

export default muiThemeable()(AlertService);
