// way2go-common/src/components/TooltipPopover/TooltipPopover.js
import React from 'react';
import Popover from 'material-ui/Popover';
import muiThemeable from 'material-ui/styles/muiThemeable';

class TooltipPopover extends React.Component {
  constructor(props) {
    super(props);

    this.interval = null;

    this.state = {open: false};

    this.handleRequestClose = this.handleRequestClose.bind(this);
  }

  componentDidMount(){

  }

  componentWillReceiveProps(nextProps) {

  }

  showPopover = (e) => {
    this.setState({anchorEl: e.target, open: true});

    clearInterval(this.interval);
    this.interval = setInterval(this.handleRequestClose, 5000);
  }

  handleRequestClose() {
    this.setState({open: false});
    if(this.props.handleRequestClose)
      this.props.handleRequestClose();
  };

  render() {
    let arrowUp = {
      marginLeft: 'auto',
      marginRight: 'auto',
      width: 0,
      height: 0,
      borderLeft: '5px solid transparent',
      borderRight: '5px solid transparent',
      borderBottom: '5px solid #21658a',
      borderBottomColor: this.props.muiTheme.palette.primary2Color
    }
    let popoverContainer = {
      marginBottom: '6px',
      marginRight: '2px',
      marginLeft: '2px',
      padding: '5px 10px',
      boxShadow: 'rgba(0, 0, 0, 0.12) 0px 1px 6px, rgba(0, 0, 0, 0.12) 0px 1px 4px',
      borderColor: this.props.muiTheme.palette.primary2Color,
      backgroundColor: this.props.muiTheme.palette.primary2Color,
      borderWidth: '1px',
      borderStyle: 'solid',
      borderRadius: '5px',
      color: '#FFF',
      fontSize: '12px'
    }
    return (
      <div style={{display: 'inline'}}>
        <span
          onMouseEnter={(e) => this.showPopover(e)}
          onMouseOver={(e) => this.showPopover(e)}
          onMouseMove={(e) => this.showPopover(e)}
          onMouseOut={(e) => this.handleRequestClose()}>
          {this.props.link}
        </span>
        {
          this.state.open &&
          <Popover open={this.state.open} onRequestClose={this.handleRequestClose}  anchorEl={this.state.anchorEl} useLayerForClickAway={false} style={{backgroundColor: 'transparent', boxShadow: 'none'}} anchorOrigin = {{"horizontal":"middle", "vertical":"bottom"}} targetOrigin = {{"horizontal":"middle", "vertical":"top"}}>
            <div class="arrow_up" style={arrowUp}></div>
            <div style={popoverContainer}>
              {this.props.children}
            </div>
          </Popover>
        }
      </div>
    );
  }
}

export default muiThemeable()(TooltipPopover);
