// way2go-authentication/lib/components/ForgotPassword/forgotPwdComponent.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactDom = require('react-dom');

var _reactDom2 = _interopRequireDefault(_reactDom);

var _TextField = require('material-ui/TextField');

var _TextField2 = _interopRequireDefault(_TextField);

var _RaisedButton = require('material-ui/RaisedButton');

var _RaisedButton2 = _interopRequireDefault(_RaisedButton);

var _Snackbar = require('material-ui/Snackbar');

var _Snackbar2 = _interopRequireDefault(_Snackbar);

var _muiThemeable = require('material-ui/styles/muiThemeable');

var _muiThemeable2 = _interopRequireDefault(_muiThemeable);

var _way2goCommon = require('way2go-common');

var _forgotPwdApi = require('./forgotPwdApi');

var _forgotPwdApi2 = _interopRequireDefault(_forgotPwdApi);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var snackBarHeight = {
    height: "70%"
};

var ForgotPassword = function (_React$Component) {
    _inherits(ForgotPassword, _React$Component);

    function ForgotPassword(props) {
        _classCallCheck(this, ForgotPassword);

        var _this = _possibleConstructorReturn(this, (ForgotPassword.__proto__ || Object.getPrototypeOf(ForgotPassword)).call(this, props));

        _this.saveChanges = function (user) {
            _forgotPwdApi2.default.forgotPwd(user).then(function (response) {
                if (response.data.message === "success") {
                    _this.setState({ statusMessage: response.data.data, showAlert: true });
                } else {
                    _this.setState({ statusMessage: response.data.message, showAlert: true });
                }
            });
        };

        _this.handleSubmit = function () {
            if (_this.state.userName.length == 0) {
                _this.setState({ errorMsgUserName: "This field is required" });
            }
            if (_this.state.email.length == 0) {
                _this.setState({ errorMsgEmail: "This field is required" });
            }
            if (_this.state.userName.length != 0 && _this.state.email.length != 0) {
                var atpos = _this.state.email.indexOf("@");
                var dotpos = _this.state.email.lastIndexOf(".");
                if (atpos < 1 || dotpos < atpos + 2 || dotpos + 2 >= _this.state.email.length) {
                    _this.setState({ errorMsgEmail: "Not a valid e-mail address" });
                } else {
                    _this.setState({ errorMsgEmail: "" });
                    var user = {};
                    user.orgId = (0, _way2goCommon.Storage)().getValue((0, _way2goCommon.Storage)().options.ORG_ID).toString();
                    user.userName = _this.state.userName;
                    user.email = _this.state.email;
                    _this.saveChanges(user);
                }
            }
        };

        _this.handleuserName = function (e) {
            _this.setState({ userName: e.target.value });
            if (e.target.value.length == 0) {
                console.log("length", e.target.value.length);
                _this.setState({ errorMsgUserName: "This field is required" });
            } else {
                _this.setState({ errorMsgUserName: "" });
            }
        };

        _this.handleemailId = function (e) {
            _this.setState({ email: e.target.value });
            if (e.target.value.length == 0) {
                _this.setState({ errorMsgEmail: "This field is required" });
            } else {
                _this.setState({ errorMsgEmail: "" });
            }
        };

        _this.state = {
            statusMessage: "",
            userName: "",
            email: "",
            errorMsgUserName: "",
            errorMsgEmail: "",
            showAlert: false,
            open: true
        };
        _this.handleuserName.bind(_this);
        return _this;
    }
    //function to hit th api and get the response


    _createClass(ForgotPassword, [{
        key: 'render',
        value: function render() {
            return _react2.default.createElement(
                'div',
                null,
                _react2.default.createElement(_TextField2.default, { onChange: this.handleuserName, errorText: this.state.errorMsgUserName, name: 'userName', id: 'userName', floatingLabelText: 'User Name', autoFocus: true, fullWidth: true }),
                _react2.default.createElement('br', null),
                _react2.default.createElement(_TextField2.default, { onChange: this.handleemailId, errorText: this.state.errorMsgEmail, name: 'email', id: 'email', floatingLabelText: 'Email Id', fullWidth: true }),
                _react2.default.createElement('br', null),
                _react2.default.createElement('br', null),
                _react2.default.createElement(
                    'div',
                    { style: { padding: '8px 16px' } },
                    _react2.default.createElement(_RaisedButton2.default, { label: 'Submit', primary: true, onClick: this.handleSubmit, fullWidth: true })
                ),
                _react2.default.createElement('br', null),
                _react2.default.createElement(
                    'div',
                    { style: { cursor: 'pointer' }, onClick: function onClick() {
                            document.getElementById("loginComponent").style.display = 'block';document.getElementById("forgotPwdComponent").style.display = 'none';
                        } },
                    _react2.default.createElement(
                        'u',
                        null,
                        'Back to Login'
                    )
                ),
                _react2.default.createElement('br', null),
                _react2.default.createElement(_Snackbar2.default, {
                    open: this.state.showAlert,
                    message: this.state.statusMessage,
                    autoHideDuration: 4000,
                    bodyStyle: snackBarHeight
                })
            );
        }
    }]);

    return ForgotPassword;
}(_react2.default.Component);

exports.default = (0, _muiThemeable2.default)()(ForgotPassword);
