// way2go-authentication/src/components/Login/LoginComponent.js
import React from 'react';
import PropTypes from 'prop-types';
import {purple500} from 'material-ui/styles/colors';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';
import Paper from 'material-ui/Paper';
import Container from 'muicss/lib/react/container';
import Row from 'muicss/lib/react/row';
import Col from 'muicss/lib/react/col';
// import {errormsg} from '../../css/style.css';
// import {errormsg} from '../../css/style.css';
import web_logoback from '../../assets/images/web_logoback.png';
import login_web from '../../assets/images/login_web.jpg';
import Logo from './Logo';
import ForgotPassword from '../ForgotPassword/forgotPwdComponent';

const muiTheme = getMuiTheme({
  palette: {
    primary1Color: purple500,
  },
});


const style = {
  padding: 20,
  display: 'inline-block',
  borderRadius: '6px',
  border: '1px solid #dbdbdb',
  backgroundColor: 'rgb(245, 245, 245)',
  width: '263px',
  position: 'relative',
  marginTop: '-133px',
  paddingTop: '125px'
};

const backgroundStyle = {
  backgroundImage: 'url("' + login_web + '")',
  WebkitBackgroundSize: 'cover',
  height: '100%'
};
const mainDivStyle={
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    height: '100%'
};
const logoDivStyle={
    width: '285px',
    backgroundImage: 'url("' + web_logoback + '")',
    position: 'relative',
    zIndex: 150,
    height: '77px',
    marginTop: '56px',
    marginLeft: 'auto',
    marginRight: 'auto'
};
export const LoginComponent = ({loginObj, loginSubmit, loginResponseObj, onChange, showLoggedOut}) => {

    return(
        <div style={backgroundStyle}>
            <Container fluid={true} >
                <Row style={mainDivStyle} >
                    <Col md-offset="6" md="6" sm-offset="4" sm="8" className="mui--text-center">
                        <div style={logoDivStyle}>
                            <Logo />
                        </div>
                        <MuiThemeProvider muiTheme={muiTheme}>
                            <Paper style={style} zDepth={1}>
                                <form id="loginComponent">
                                    <TextField name="userName" floatingLabelText="Username" defaultValue={loginObj.userName} onChange={onChange} fullWidth={true} autoFocus={true} />
                                    <br />
                                    <TextField name="password" floatingLabelText="Password" defaultValue={loginObj.password} type="password" onChange={onChange} fullWidth={true} />
                                    <br />
                                    <br />
                                    {showLoggedOut && <div className="mui--text-caption" style={{color: 'red'}}><br />You have logged out successfully<br /><br /></div>}
                                    <div style={{padding: '8px 16px'}}>
                                        <RaisedButton type="submit" label="Log In" primary onClick={loginSubmit} fullWidth={true}/>
                                    </div>
                                    <br />
                                    <div style={{cursor:'pointer'}} onClick={function(){document.getElementById("loginComponent").style.display='none';document.getElementById("forgotPwdComponent").style.display='block';document.getElementById("email").value='';document.getElementById("userName").value='';}}><u>Forgot Password</u></div>

                                    <br /> { loginResponseObj.outputCode == 500 ?
                                    <ShowError error={loginResponseObj}/> : <span>&nbsp;</span> }
                                </form>
                                <div id="forgotPwdComponent" style={{display:'none'}}>
                                    <ForgotPassword />
                                </div>
                            </Paper>
                        </MuiThemeProvider>
                    </Col>
                </Row>
            </Container>
        </div>
    );
};

LoginComponent.propTypes={
    loginObj: PropTypes.object.isRequired,
    loginSubmit: PropTypes.func.isRequired,
    onChange: PropTypes.func.isRequired,
    loginResponseObj: PropTypes.object.isRequired,
    onFPChange: PropTypes.func.isRequired,
};

const ShowError = ({error}) => {
    return(
        <div className="errormsg">
            {error.message}
        </div>
    );
};

ShowError.propTypes={
    error: PropTypes.object.isRequired,
};

export default LoginComponent;
