// way2go-authentication/src/components/Login/LoginActions.js
//export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';

import LoginApi from './LoginApi';
import ThemesFetchApi from './ThemesFetchApi';

export function loginSuccess(loginResponse) {
    return { type: 'LOGIN_SUCCESS', loginResponse: {loginResponse: loginResponse.data} };
}

export function themesSuccess(themesResponse) {
    return { type: 'THEMES_SUCCESS', themesResponse: {themesResponse: themesResponse.data } };
}

function authenticationModeSuccess(response) {
  return {
    type: 'AUTH_MODE_SUCCESS',
    response: response
  }
}

export function getAuthenticationMode(orgId) {
  return function(dispatch) {
    LoginApi.authenticateService(orgId).then(result => {
    if(result.data.outputCode == 200) {
      dispatch(authenticationModeSuccess(result.data.data));
    }
  });
  }
}

export function login(loginObj) {
    return function (dispath) {
        return LoginApi.login(loginObj).then(result => {
            dispath(loginSuccess(result));
        }).catch(error => {
            throw (error);
        });
    };
}

export function themesFetch(reqObj) {
    return function (dispath) {
        return ThemesFetchApi.themesFetch(reqObj).then(result => {
            dispath(themesSuccess(result));
        }).catch(error => {
            throw (error);
        });
    };
}
